docs/architecture/adr-017-single-binary-upgrade.md

```diff
@@ -0,0 +1,141 @@
+# ADR 017: Single binary upgrade
+
+## Status
+
+Proposed
+
+## Changelog
+
+- 2023/03/22: initial draft
+
+## Context
+
+It important to sync from genesis with one binary.
+It is needed for wallets, block explorers, indexers, analytics (e.g. Nansen).
+It makes it possible to sync headers from chain and download blobs from third party service.
```

Comment:
I think we need to clarify this. In the meeting the point was made that the Celestia network doesn't need to retain all historical blob data and that blobs could eventually be pruned and fetched from a third party service.

I don't see how single binary sync inherently makes this possible. Maybe this can state that an engineering requirement is that a single celestia-app binary can fetch all historical block headers from the chain (?)

Reply:
bumping this :slightly_smiling_face: